core/system/src/pool_queue.c

```diff
@@ -361,10 +361,11 @@ static uvisor_pool_slot_t find_first(uvisor_pool_queue_t * pool_queue,
 {
     uvisor_pool_slot_t slot;
     uvisor_pool_t * pool = UVISOR_AUTO_NS_ALIAS(pool_queue->pool);
+    uvisor_pool_slot_t iterated = 0;
 
     /* Walk the queue, looking for the first slot that matches the query. */
     slot = pool_queue->head;
-    while (slot != UVISOR_POOL_SLOT_INVALID)
+    while (slot != UVISOR_POOL_SLOT_INVALID && iterated <= pool->num)
```

Comment:
It wouldn't make much of a different because uint8_MAX is 255 whereas MAX_VALID is 252, but I'll do it anyways